src/cursor.c

```diff
@@ -180,54 +176,54 @@ static uint64_t CursorList_GenerateId(CursorList *curlist) {
   return id;
 }
 
-Cursor *Cursors_Reserve(CursorList *cl, const char *lookupName, unsigned interval,
+Cursor *Cursors_Reserve(CursorList *cl, StrongRef global_spec_ref, unsigned interval,
                         QueryError *status) {
   CursorList_Lock(cl);
   CursorList_IncrCounter(cl);
-  CursorSpecInfo *spec = findInfo(cl, lookupName);
   Cursor *cur = NULL;
 
-  if (spec == NULL) {
-    QueryError_SetErrorFmt(status, QUERY_ENOINDEX, "Index `%s` does not have cursors enabled",
-                           lookupName);
-    goto done;
-  }
+  // If the cursor should be associated with a spec, 
+  // we assume that global_spec_ref points to a valid spec, else the function returns NULL.
+  IndexSpec *spec = StrongRef_Get(global_spec_ref);
 
-  if (spec->used >= spec->cap) {
+  // If we are in a coordinator ctx, the spec is NULL
+  if (spec && (spec->activeCursors >= spec->cursorsCap)) {
     /** Collect idle cursors now */
     Cursors_GCInternal(cl, 0);
-    if (spec->used >= spec->cap) {
+    if (spec->activeCursors >= spec->cursorsCap) {
```

Comment:
you need to demote strong ref here

Reply:
Note that this function gets a strong refernce, so it is assumed that the management of the reference is done by the calling function.